app/src/main/java/org/mozilla/fenix/historymetadata/controller/HistoryMetadataController.kt

```diff
@@ -53,6 +62,13 @@ class DefaultHistoryMetadataController(
         )
     }
 
+    override fun handleRemoveGroups(groups: Set<HistoryMetadataGroup>) {
```

Comment:
`handleRemoveGroup(group: HistoryMetadataGroup)`

This _could_ just take searchTerm and be named something like `handleRemoveSearchGroup`, but in the near future, we'll have other types of groups (referrers, from external apps, etc), and they'll all have distinct deletion logic provided by the storage API. By keeping this handler generic we're making it easy to match on group types in the future and call an appropriate method.